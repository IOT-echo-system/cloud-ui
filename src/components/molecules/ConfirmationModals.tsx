import type {PropsWithChildren} from 'react'
import React, {useState} from 'react'
import {Stack} from '@mui/material'
import type { ConfirmationModalPropsType} from '../atoms'
import {ConfirmationModal as ConfirmationModalUnit} from '../atoms'

export type GetConfirmationModalPropsTypeFunction<T extends Record<string, unknown> = Record<string, unknown>> = (
  handleClose: () => void,
  props: T
) => ConfirmationModalPropsType

type ConfirmationModalsPropsType<T extends Record<string, unknown>> = {
  getConfirmationModalDetails: GetConfirmationModalPropsTypeFunction<T>
} & T

export const ConfirmationModals = <T extends Record<string, unknown>>(
  props: PropsWithChildren<ConfirmationModalsPropsType<T>>
): React.JSX.Element => {
  const {children, getConfirmationModalDetails} = props
  const [open, setOpen] = useState(false)
  const handleClose = () => {
    setOpen(false)
  }

  const {title, description, loading, cancelText, confirmText, onCancel, onConfirm, cancelColor, confirmColor} =
    getConfirmationModalDetails(handleClose, props)

  return (
    <Stack>
      <Stack
        onClick={() => {
          setOpen(true)
        }}
      >
        {children}
      </Stack>
      <ConfirmationModalUnit
        open={open}
        handleClose={handleClose}
        title={title}
        description={description}
        loading={loading}
        cancelText={cancelText}
        confirmText={confirmText}
        cancelColor={cancelColor}
        confirmColor={confirmColor}
        onConfirm={onConfirm}
        onCancel={onCancel}
      />
    </Stack>
  )
}
