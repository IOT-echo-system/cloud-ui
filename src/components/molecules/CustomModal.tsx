import type {PropsWithChildren} from 'react'
import {useState} from 'react'
import React from 'react'
import {Stack} from '@mui/material'
import {Modal} from '../atoms'

type CustomModalPropsType = {ClickableComponent: React.JSX.Element}

export const CustomModal: React.FC<PropsWithChildren<CustomModalPropsType>> = ({children, ClickableComponent}) => {
  const [open, setOpen] = useState(false)
  const handleClose = () => {
    setOpen(false)
  }

  return (
    <Stack>
      <Stack
        sx={{cursor: 'pointer'}}
        onClick={() => {
          setOpen(true)
        }}
      >
        {ClickableComponent}
      </Stack>
      <Modal open={open} handleClose={handleClose}>
        {children}
      </Modal>
    </Stack>
  )
}
