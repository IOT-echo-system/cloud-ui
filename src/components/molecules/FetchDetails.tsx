import type {PropsWithChildren} from 'react'
import React, {useEffect} from 'react'
import {useDispatch, useSelector} from '../../hooks'
import {BoardService} from '../../services/boardService'
import {setBoards} from '../../store/actions/boards'

export const FetchDetails: React.FC<PropsWithChildren> = ({children}) => {
  const dispatch = useDispatch()
  const {user, project} = useSelector(state => state)

  useEffect(() => {
    if (user.userId && project.projectId) {
      BoardService.getBoards().then(boards => {
        dispatch(setBoards(boards))
      })
    }
  }, [user.userId, project.projectId])

  return <>{children}</>
}
