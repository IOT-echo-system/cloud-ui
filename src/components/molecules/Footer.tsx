import {Stack} from '@mui/material'
import React from 'react'
import {CopyRights, PageContainer} from '../atoms'

export const Footer: React.FC = () => {
  return (
    <Stack sx={{padding: {xs: '16px 0', md: '24px 0', lg: '32px 0'}}}>
      <PageContainer>
        <CopyRights />
      </PageContainer>
    </Stack>
  )
}
