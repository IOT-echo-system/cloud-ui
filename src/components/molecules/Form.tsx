import {Stack, Typography} from '@mui/material'
import type {FormEvent} from 'react'
import React from 'react'
import type {FormInputType} from '../atoms'
import {Button, FormContainer, FormInput} from '../atoms'

export type FormProps = {
  handleSubmit: (event: FormEvent<HTMLFormElement>) => void
  title: string
  inputFields: FormInputType[]
  submitBtnText: string
  submitBtnDisabled?: boolean
  loading?: boolean
}

export const Form: React.FC<FormProps> = props => {
  return (
    <FormContainer onSubmit={props.handleSubmit}>
      <Stack spacing={2}>
        <Typography variant={'h5'}>{props.title}</Typography>
        {props.inputFields.map((inputField, index) => (
          <FormInput key={`input-${index}`} {...inputField} />
        ))}
        <Button
          loading={props.loading ?? false}
          type={'submit'}
          variant={'contained'}
          size={'large'}
          disabled={props.submitBtnDisabled}
        >
          {props.submitBtnText}
        </Button>
      </Stack>
    </FormContainer>
  )
}
