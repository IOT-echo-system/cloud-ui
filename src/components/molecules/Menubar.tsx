import React from 'react'
import {List, ListItem, Stack, styled} from '@mui/material'
import {Link} from '../atoms'
import {useRouter} from 'next/router'
import type {LinkProps} from 'next/link'

const StyledLink = styled(Link)<LinkProps & {active: boolean}>(({theme, active}) => ({
  color: theme.palette.common.white,
  fontWeight: active ? 'bold' : 'normal',
  '&:hover': {
    textDecoration: 'none'
  }
}))

const menuItems = [
  {name: 'Home', link: '/'},
  {name: 'Dashboard', link: '/dashboard'}
]

export const Menubar: React.FC = () => {
  const router = useRouter()

  return (
    <Stack direction={'row'} component={List}>
      {menuItems.map(({name, link}) => (
        <ListItem key={link}>
          <StyledLink href={link} active={router.pathname.startsWith(link)}>
            {name}
          </StyledLink>
        </ListItem>
      ))}
    </Stack>
  )
}
