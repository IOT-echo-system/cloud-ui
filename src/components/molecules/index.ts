export * from './Header/Header'
export * from './Footer'
export * from './Form'
export * from './ValidatedProfile'
export * from './FetchDetails'
export * from './CustomModal'
export * from './BoardSecretKey/BoardSecretKey'
export * from './ConfirmationModals'
export * from './MqttSubscriber'
