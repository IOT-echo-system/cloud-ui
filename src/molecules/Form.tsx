import {Stack, Typography} from '@mui/material'
import type {FormEvent} from 'react'
import React from 'react'
import type {FormInputType} from '@/atoms'
import {Button, FormContainer, FormInput} from '@/atoms'

export type FormProps = {
  handleSubmit: (event: FormEvent<HTMLFormElement>) => void
  title: string
  error?: string
  inputFields: FormInputType[]
  submitBtnText: string
  submitBtnDisabled?: boolean
}

export const Form: React.FC<FormProps> = props => {
  return (
    <FormContainer onSubmit={props.handleSubmit}>
      <Stack spacing={2}>
        <Typography variant={'h5'}>{props.title}</Typography>
        {props.error && (
          <Typography variant={'body1'} color={'error'}>
            {props.error}
          </Typography>
        )}
        {props.inputFields.map((inputField, index) => (
          <FormInput key={`input-${index}`} {...inputField} />
        ))}
        <Button type={'submit'} variant={'contained'} size={'large'} disabled={props.submitBtnDisabled}>
          {props.submitBtnText}
        </Button>
      </Stack>
    </FormContainer>
  )
}
