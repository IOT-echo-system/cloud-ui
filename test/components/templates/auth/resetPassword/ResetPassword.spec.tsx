import React from 'react'
import {render} from '@testing-library/react'
import * as UseResetPasswordHook from '../../../../../src/components/templates/auth/resetPassword/useResetPassword'
import {ResetPassword} from '../../../../../src/components/templates/auth'
import {Form} from '../../../../../src/components/molecules/Form'

jest.mock('../../../../../src/components/molecules/Form')
jest.mock('../../../../../src/components/templates/auth/resetPassword/useResetPassword')

describe('Reset Password Template Test', () => {
  const mockUseReset = {inputFields: [], error: '', handleSubmit: jest.fn()}

  beforeEach(() => {
    jest.resetAllMocks()
    jest.spyOn(UseResetPasswordHook, 'useResetPassword').mockReturnValue(mockUseReset)
  })
  afterEach(jest.resetAllMocks)

  it('should match with the snapshot', () => {
    const {container} = render(<ResetPassword title={'Change password'} withOldPassword={false} redirectTo={'/'} />)

    expect(UseResetPasswordHook.useResetPassword).toHaveBeenCalledTimes(1)
    expect(container).toMatchSnapshot()
    expect(Form).toHaveBeenCalledTimes(1)
    expect(Form).toHaveBeenCalledWith(
      {
        error: '',
        handleSubmit: expect.any(Function),
        inputFields: [],
        submitBtnText: 'Reset password',
        title: 'Change password'
      },
      {}
    )
  })

  it('should match with the snapshot with old password', () => {
    const {container} = render(<ResetPassword title={'Change password'} withOldPassword redirectTo={'/'} />)

    expect(UseResetPasswordHook.useResetPassword).toHaveBeenCalledTimes(1)
    expect(container).toMatchSnapshot()
    expect(Form).toHaveBeenCalledTimes(1)
    expect(Form).toHaveBeenCalledWith(
      {
        error: '',
        handleSubmit: expect.any(Function),
        inputFields: [],
        submitBtnText: 'Reset password',
        title: 'Change password'
      },
      {}
    )
  })
})
