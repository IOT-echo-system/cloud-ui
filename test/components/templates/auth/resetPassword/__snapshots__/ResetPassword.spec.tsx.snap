// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password Template Test should match with the snapshot 1`] = `<div />`;

exports[`Reset Password Template Test should match with the snapshot with old password 1`] = `<div />`;
